### DISCUSSION TITLE
How to handle node server tokens securely in refine

### DISCUSSION THREAD

**Discussion by van-vince** [refinedev/refine#5116] [Q&A] [answered]
I want to know how to securely save nodejs tokens in my refine dashboard without saving in localstorage. I am using headless auth provider and all the examples i have seen saves tokens in local storage. Also how can I make requests with credentials.

**Comment by BatuhanW** [comment, marked as answer]
Hello @van-vince, you can use cookies, we have some docs with NextJS and remix:

https://refine.dev/docs/packages/documentation/routers/nextjs/#server-side

https://refine.dev/docs/packages/documentation/routers/remix/#server-side-authentication-with-self-service-cookie


This example uses localstorage, in your case you can read from cookies. 
https://refine.dev/docs/tutorial/understanding-authprovider/create-authprovider/#setting-authorization-credentials
